instrumentation/failsafe-3.0/library/src/main/java/io/opentelemetry/instrumentation/failsafe/v3_0/CircuitBreakerEventListenerBuilders.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.failsafe.v3_0;
+
+import dev.failsafe.CircuitBreakerConfig;
+import dev.failsafe.event.CircuitBreakerStateChangedEvent;
+import dev.failsafe.event.EventListener;
+import dev.failsafe.event.ExecutionCompletedEvent;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.metrics.LongCounter;
+import io.opentelemetry.api.metrics.Meter;
+
+final class CircuitBreakerEventListenerBuilders {
+  private CircuitBreakerEventListenerBuilders() {}
+
+  static <R> EventListener<ExecutionCompletedEvent<R>> buildInstrumentedFailureListener(
+      CircuitBreakerConfig<R> userConfig, Meter meter, Attributes attributes) {
+    LongCounter failureCounter =
+        meter
+            .counterBuilder("failsafe.circuit_breaker.failure.count")
```

Comment:
maybe @trask  can weigh in on this too because I'm not sure if it's directly applicable, but the docs around "[recording errors on metrics](https://opentelemetry.io/docs/specs/semconv/general/recording-errors/#recording-errors-on-metrics)" recommends using a single metric with an attribute that differentiates between success and failure

> It’s RECOMMENDED to report one metric that includes successes and failures as opposed to reporting two (or more) metrics depending on the operation status.